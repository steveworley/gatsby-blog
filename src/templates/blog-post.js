import React, { Component } from "react"
import { graphql } from "gatsby"
import Layout from "../components/layout"
import { Helmet } from "react-helmet"

import styles from './blog-post.module.css'

class BlogPost extends Component {
  render() {
    const post = this.props.data.nodeBlog
    const comments = this.props.data.allCommentComment ? this.props.data.allCommentComment.edges : []

    return (
      <Layout>
        <Helmet>
          <meta charSet="utf-8" />
          <title>{ post.title } - Steve Worley</title>
          <link rel="canonical" href={"http://steveworley.github.io." + post.fields.slug } />
        </Helmet>
        <div className={styles['post--content']}>
          <h1>{post.title}</h1>
          <div dangerouslySetInnerHTML={{ __html: post.body.processed }} />
        </div>
        <div className={styles.postComments}>
          { comments.map(({node}) => (
            <div key={node.id} className={styles.postComment}>
              <h3 className={styles.postCommentTitle}>{ node.subject }</h3>
              <div dangerouslySetInnerHTML={{ __html: node.comment_body.processed }} />
            </div>
          ))}
        </div>
      </Layout>        
    )
  }
}

export default BlogPost

export const query = graphql`
  query($nid: Int!, $filter: filterCommentComment) {
    nodeBlog(nid: { eq: $nid } ) {
      title
      nid
      body {
        processed
      }
      fields {
        slug
      }
      relationships {
        field_tags {
          id
          name
          fields {
            slug
          }
        }
      }
    }
    allCommentComment(filter:$filter){
      totalCount
      edges{
        node{
          subject
          comment_body{
            processed
          }
          id
          relationships{
            entity_id{
              id
            }
          }
        }
      }
    }
  }
`