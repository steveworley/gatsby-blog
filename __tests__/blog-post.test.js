import React from 'react'
import renderer from 'react-test-renderer'
import BlogPost from '../src/templates/blog-post'

import { withComments, withoutComments } from '../__mocks__/blog-post.mock'

jest.mock('../src/components/layout',  () => 'Layout')

describe('Blog post', () => {
  it ('renders correctly with comments', () => {
    const tree = renderer.create(<BlogPost data={withComments} />).toJSON()
    expect(tree).toMatchSnapshot();
  })

  it ('renders correctly without comments', () => {
    const tree = renderer.create(<BlogPost data={withoutComments} />).toJSON()
    expect(tree).toMatchSnapshot();
  })
})