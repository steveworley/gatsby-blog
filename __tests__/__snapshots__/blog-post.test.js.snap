// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog post renders correctly with comments 1`] = `
<Layout>
  <div
    className="post--content"
  >
    <h1>
      Youtube z-index fix
    </h1>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Flash appearing behind elements is a common problem with a relatively simple fix — simply add wmode=transparent to the object tag and your Flash movie will behave nicely with the CSS z-index property.</p>

<p>But what happens when you embed a YouTube video?</p>

<p>YouTube’s relatively new embedding code utilizes iFrames — this means that we can’t simply add the wmode tag to the flash object. Here is a little trick to make YouTube include videos that respect z-indexing.</p>

<p>Here is our regular embed code.</p>

<pre>
<code>
&lt;iframe width=\\"420\\" height=\\"315\\" src=\\"http://www.youtube.com/embed/C4I84Gy-cPI\\" frameborder=\\"0\\" allowfullscreen&gt;</code></pre>

<p>We can simply add \`?wmode=transparent\` to the end of YouTube URL. This will tell YouTube to include the video with the wmode set. So you new embed code will look like this;</p>

<pre><code>&lt;iframe width=\\"420\\" height=\\"315\\" src=\\"http://www.youtube.com/embed/C4I84Gy-cPI?wmode=transparent\\" frameborder=\\"0\\" allowfullscreen&gt;</code></pre>
",
        }
      }
    />
  </div>
  <div
    className="postComments"
  >
    <div
      className="postComment"
    >
      <h3
        className="postCommentTitle"
      >
        Test
      </h3>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>This is a test comment.</p>
",
          }
        }
      />
    </div>
    <div
      className="postComment"
    >
      <h3
        className="postCommentTitle"
      >
        Second comment!
      </h3>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "#2!",
          }
        }
      />
    </div>
  </div>
</Layout>
`;

exports[`Blog post renders correctly without comments 1`] = `
<Layout>
  <div
    className="post--content"
  >
    <h1>
      CSS attribute Selectors
    </h1>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>An interesting and somewhat unused feature of CSS is that it allows you to target elements based on the attributes of those elements. You are probably familiar with the more common ones; like type when trying to select different forms of input elements, however you can actually target quite a few other attributes in this way, modern browsers will allow you to select any attribute name however it will only work in \`IE7\` and \`IE8\` if a \`!DOCTYPE\` is specified. You can also use operational selectors (not just equals) similar to jQuery. These operators include:</p>

<ul><li>*= contains</li>
	<li>~= contains word</li>
	<li>^= starts with</li>
	<li>$=ends with</li>
</ul><p>Unfortunately we cannot do “not equals” (!=) with the attribute selectors in CSS. You can however mix two together to get a pseudo not equals to ie. \`[rel=\\"this\\"][rel=\\"this2\\"]\` and this will only select elements that match both.</p>

<p>I have tried the following in IE7+.</p>

<ul><li>[data-*=\\"val\\"]</li>
	<li>[rel=\\"val\\"]</li>
	<li>[title=\\"val\\"]</li>
	<li>[href=\\"val\\"]</li>
	<li>[aria-*=\\"val\\"]</li>
</ul><p>You can even use custom attribute selectors; like \`[myattr=\\"val\\"]\`.</p>

<p>For example lets say we wanted to target an iframe that Google AdWords includes for conversion tracking; unfortunately the iframe does not have a class or an ID — no matter we can select it from its name attribute. Take the given iframe</p>

<pre>
<code>&lt;iframe name=\\"google_conversion_tracking\\" src=\\"...\\"&gt;&lt;/iframe&gt;</code></pre>

<p>We can select it in CSS with</p>

<pre>
<code>
iframe[name=\\"google_conversion_tracking\\"] {
   /* your styles */
}
</code></pre>
",
        }
      }
    />
  </div>
  <div
    className="postComments"
  />
</Layout>
`;
